pkg/config/config.go

```diff
@@ -95,6 +95,20 @@ func getLocalConfigFile(cfgDir string) (string, error) {
 	return filepath.Join(cfgDir, ".odo", configFileName), nil
 }
 
+// LocalConfigExists checks if a config file exists.
+// If not it returns false
+func LocalConfigExists(cfgDir string) (bool, error) {
```

Comment:
I was thinking if it is better instead to add a bool flag to indicate if NewLocalConfigInfo and/or New should return an error if config does not exist it might be ok to even have it as a optional arguement(variable length args) with the current behaviour of those functions as default ?

Reply:
Possible maybe. But having a specialized function that does not create but rather just checks if it exists helps. Overloading one function with too much functionality could become a problem. Also, this might end up being a fix for now as component list is not yet tied to `--context`, so both these cases will need to be covered ie it checks for `--context` as well.

Reply:
Well IMO, this function is actually replicating initial logic from NewLocalConfigInfo except 1 conditional handling which can be easily controlled by a bool.

Reply:
Alright i will take a look :)

Reply:
Updated